File: PlacementCellBackend/Services/CRUD/Interfaces/ITeacherService.cs
using PlacementCellBackend.Models;

namespace PlacementCellBackend.Services.CRUD.Interfaces
{
    public interface ITeacherService
    {
        Task<IEnumerable<Teacher>> GetAllTeachersAsync();
        Task<Teacher?> GetTeacherByIdAsync(string id);
        Task<Teacher> CreateTeacherAsync(Teacher teacher);
        Task<bool> UpdateTeacherAsync(string id, Teacher teacher);
        Task<bool> DeleteTeacherAsync(string id);
        bool TeacherExists(string id);
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/services/Interfaces/ITeacherService.java
package com.example.placementicsbackend.services.Interfaces;

import com.example.placementicsbackend.models.Teacher;
import java.util.List;

public interface ITeacherService {

    List<Teacher> getAllTeachers();

    Teacher getTeacherById(String id);

    Teacher createTeacher(Teacher teacher);

    boolean updateTeacher(String id, Teacher teacher);

    boolean deleteTeacher(String id);

    boolean teacherExists(String id);
}
